pkg/apis/garden/types.go

```diff
@@ -1240,3 +1247,98 @@ const (
+type BackupInfrastructureList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta
+	// Items is the list of BackupInfrastructure.
+	Items []BackupInfrastructure
+}
+
+// BackupInfrastructureSpec is the specification of a Backup Infrastructure.
+type BackupInfrastructureSpec struct {
+	// Seed is the name of a Seed object.
+	Seed string
+	// GracePeriod holds the time to leave in number of days the Backup Infrastructure after deletion timestamp is set.
+	// +optional
+	GracePeriod *int
+}
+
+// BackupInfrastructureStatus holds the most recently observed status of the Backup Infrastructure.
+type BackupInfrastructureStatus struct {
+	// Gardener holds information about the Gardener which last acted on the Backup Infrastructure.
+	Gardener Gardener
+	// Phase holds information about the last operation on the Backup Infrastructure.
+	// +optional
+	Phase *BackupInfrastructurePhase
+	// LastError holds information about the last occurred error during an operation.
+	// +optional
+	LastError *LastError
+	// ObservedGeneration is the most recent generation observed for this BackupInfrastructure. It corresponds to the
+	// BackupInfrastructure's generation, which is updated on mutation by the API Server.
+	// +optional
+	ObservedGeneration *int64
+	// RetryCycleStartTime is the start time of the last retry cycle (used to determine how often an operation
+	// must be retried until we give up).
+	// +optional
+	RetryCycleStartTime *metav1.Time
+}
+
+// BackupInfrastructurePhase holds the state of operation on backup infrastructure.
+type BackupInfrastructurePhase string
+
+const (
+	// PhaseReconciling indicates that reconciliation operation is going on
+	PhaseReconciling BackupInfrastructurePhase = "Reconciling"
+	// PhaseReconciled indicates indicates that an operation has completed successfully.
+	PhaseReconciled BackupInfrastructurePhase = "Reconciled"
+	// PhaseError indicates that an operation is completed with errors and will be retried.
+	PhaseError BackupInfrastructurePhase = "Error"
+	// PhaseFailed indicates that an operation is completed with errors and won't be retried.
+	PhaseFailed BackupInfrastructurePhase = "Failed"
+	// PhaseDeleting indicates that deletion operation is going on.
+	PhaseDeleting BackupInfrastructurePhase = "Deleting"
+	// PhaseDeleted indicates that deletion operation is successful.
+	PhaseDeleted BackupInfrastructurePhase = "Deleted"
+)
+
+const (
+	// BackupInfrastructureEventReconciling indicates that the a Reconcile operation started.
+	BackupInfrastructureEventReconciling = "ReconcilingBackupInfrastructure"
+	// BackupInfrastructureEventReconciled indicates that the a Reconcile operation was successful.
+	BackupInfrastructureEventReconciled = "ReconciledBackupInfrastructure"
+	// BackupInfrastructureEventReconcileError indicates that the a Reconcile operation failed.
+	BackupInfrastructureEventReconcileError = "ReconcileError"
```

Comment:
Why do all the other events contain "BackupInfrastructure" except this one and "DeleteError"? Please be consistent.